R/stan_module.R

```diff
@@ -41,8 +38,42 @@ read_stan <- function(string) {
     return(out)
 }
 
+#' Validate Inputs
+#'
+#' @param object
+#'
+#' @return StanModule
+#' @export StanModule
+#'
+#' @examples StanModule
+#' @name setValidity
+setValidity("StanModule",function(object){
+    if (!is.character(object@functions)
+        |!is.character(object@data)
+        |!is.character(object@parameters)
+        |!is.character(object@transformed_parameters)
+        |!is.character(object@generated_quantities))
+        stop("`Functions`, `data`, `parameters`, `transformed_parameters`
```

Comment:
from the advanced R book ( [link](https://adv-r.hadley.nz/s4.html#validator) )  I think this function is supposed to return  TRUE (if it is valid) or a string (if it isn't valid). I don't think you are supposed to call the stop function yourself, the validation takes care of that for you behind the scenes

Reply:
Hi Criag, by stop function I thought the codes may be more concise than the book's did
![image](https://user-images.githubusercontent.com/56521967/184324939-03ab51cc-d46c-4ee6-8bb4-6e26c3c8f8e7.png)
if stop function is redundant for some concern, i.e. computational cost, I will follow the book's default guide